### PR TITLE
feat: Update position when target or container dimensions change

### LINKED ISSUE #29998: Positioned elements should reposition on resize
Positioned elements should call `updatePosition` when their or their targets' dimensions change. This could be achieved by starting a resize observer once positioning is enabled on an element. 

This should remove the need to use the imperative position update for async loading content, since async content will generally resize the dimensions of the positioned element

### PR DESCRIPTION
**Pull request by ling1726**
Uses a ResizeObserver in the position manager so that a position update is triggered when the dimension of the target or container changes. This should handle most async update scenarios without needing to call `updatePosition` in userland.

The ResizeObserver will only be created when:
* both target and positioned element that refer to HTML elements
* the positionig logic is `enabled`

Practically this means that the ResizeObserver is created only when Popover/Menu/Combobox/Tooltip etc... are open. Just mounting an (unopened) Popover will not create a new ResizeObserver

Fixes #29998

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Avatar-mount_base.html">636</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Avatar-mount.html">629</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Button-mount_base.html">306</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Button-mount.html">307</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Field-mount_base.html">1123</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Field-mount.html">1148</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProvider-mount_base.html">670</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProvider-mount.html">688</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProviderWithTheme-mount.html">82</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">64</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">58</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">70</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/MakeStyles-mount_base.html">852</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/MakeStyles-mount.html">851</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Persona-mount_base.html">1738</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/Persona-mount.html">1708</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/SpinButton-mount_base.html">1388</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30179/perf-test-react-components/SpinButton-mount.html">1451</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                                                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                                                                                                                 Change |
| :---------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | -----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------: |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |   `94.95 kB`<br />`30.641 kB` |   `95.036 kB`<br />`30.68 kB` | `86 B` <br />`39 B`  |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `93.541 kB`<br />`30.319 kB` |  `93.627 kB`<br />`30.352 kB` | `86 B` <br />`33 B`  |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `209.769 kB`<br />`59.924 kB` |  `209.855 kB`<br />`59.96 kB` | `86 B` <br />`36 B`  |
| <samp>react-datepicker-compat</samp> <br /> <abbr title='bundle-size/DatePicker.fixture.js'>DatePicker Compat</abbr>                                                          | `213.412 kB`<br />`59.838 kB` | `213.497 kB`<br />`59.874 kB` | `85 B` <br />`36 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoButton.fixture.js'>InfoButton</abbr>                                                                        | `132.639 kB`<br />`41.721 kB` | `132.725 kB`<br />`41.765 kB` | `86 B` <br />`44 B`  |
| <samp>react-infobutton</samp> <br /> <abbr title='bundle-size/InfoLabel.fixture.js'>InfoLabel</abbr>                                                                          | `136.326 kB`<br />`42.862 kB` | `136.412 kB`<br />`42.903 kB` | `86 B` <br />`41 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          |  `141.96 kB`<br />`43.516 kB` | `142.046 kB`<br />`43.555 kB` | `86 B` <br />`39 B`  |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `144.646 kB`<br />`44.025 kB` | `144.732 kB`<br />`44.064 kB` | `86 B` <br />`39 B`  |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 |  `120.804 kB`<br />`38.11 kB` |  `120.89 kB`<br />`38.147 kB` | `86 B` <br />`37 B`  |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                                                                    |   `25.767 kB`<br />`9.308 kB` |   `25.858 kB`<br />`9.339 kB` | `91 B` <br />`31 B`  |
| <samp>react-timepicker-compat-preview</samp> <br /> <abbr title='bundle-size/TimePicker.fixture.js'>TimePicker</abbr>                                                         |  `96.788 kB`<br />`32.078 kB` |  `96.874 kB`<br />`32.109 kB` | `86 B` <br />`31 B`  |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `53.244 kB`<br />`18.807 kB` |   `53.33 kB`<br />`18.843 kB` | `86 B` <br />`36 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                |          Size (minified/GZIP) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                          |  `83.729 kB`<br />`23.458 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                       |  `50.167 kB`<br />`15.945 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                             |   `19.696 kB`<br />`7.795 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                     |  `64.823 kB`<br />`20.274 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ButtonProviderAndTheme.fixture.js'>react-components: Button, FluentProvider & webLightTheme</abbr> |  `69.893 kB`<br />`20.261 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>               |   `42.38 kB`<br />`14.104 kB` |
| <samp>react-persona</samp> <br /> <abbr title='bundle-size/Persona.fixture.js'>Persona</abbr>                                                                    |  `57.058 kB`<br />`17.822 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                            |    `7.099 kB`<br />`2.385 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/DataGrid.fixture.js'>DataGrid</abbr>                                                                    | `154.888 kB`<br />`43.533 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/PrimitivesOnly.fixture.js'>Table (Primitives only)</abbr>                                               |  `43.891 kB`<br />`13.782 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableAsDataGrid.fixture.js'>Table as DataGrid</abbr>                                                    | `128.109 kB`<br />`34.792 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSelectionOnly.fixture.js'>Table (Selection only)</abbr>                                            |  `73.288 kB`<br />`20.017 kB` |
| <samp>react-table</samp> <br /> <abbr title='bundle-size/TableSortOnly.fixture.js'>Table (Sort only)</abbr>                                                      |  `71.895 kB`<br />`19.617 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/InteractionTag.fixture.js'>InteractionTag</abbr>                                                         |   `15.251 kB`<br />`6.058 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/tag.fixture.js'>Tag</abbr>                                                                               |    `29.974 kB`<br />`9.44 kB` |
| <samp>react-tags</samp> <br /> <abbr title='bundle-size/TagGroup.fixture.js'>TagGroup</abbr>                                                                     |   `74.323 kB`<br />`22.28 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/167cccee21a690dd231dd3b4d4d5159cfd8169e2'>167cccee21a690dd231dd3b4d4d5159cfd8169e2</a></sub>
<div id="bundleSizeComment9423"/>

File: change/@fluentui-react-positioning-29f6eb38-4af2-40f1-a095-6abef76d64f4.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-positioning-29f6eb38-4af2-40f1-a095-6abef76d64f4.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30179&commitId=f4f436c6c581f19014419d69f55979faad7cd29a&env=prod&clientType=FLUENTUI)


  <details><summary><b>Positioning</b> 1 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Positioning.auto size with async content.floating element is within the boundary.chromium.png|3434| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

Comment (ling1726) [reply change/@fluentui-react-positioning-29f6eb38-4af2-40f1-a095-6abef76d64f4.json:1]: The autosize diff is actually OK - it's equivalent to updating that test and calling `updatePosition` after the async content has loaded

**Review by layershifter — APPROVED** [review]
Code in this PR LGTM ✅ 

The idea to have a resize observer for every popup & tooltip looked initially scary, however that statement is not correct: we will have observers only for opened popups (usually 1 or 2) & visible tooltips (should be one on a page).

**Pull request merged**